src/charm.py

```diff
@@ -634,19 +633,22 @@ def _provide_routed_ingress(self, relation: Relation):
         self._push_configurations(relation, config)
 
     def _provide_ingress(
-        self, relation: Relation, provider: Union[IngressPerAppProvider, IngressPerAppProvider]
+        self,
+        relation: Relation,
+        provider: Union[IngressPerAppProvider, IngressPerAppProvider],
```

Comment:
Elements of Union are identical.
Interesting that it wasn't picked up.

Reply:
ha.